src/components/Users/UserResetPassword.tsx

```diff
@@ -1,207 +1,217 @@
+      .string()
+      .min(1, { message: t("please_enter_current_password") }),
+    new_password_1: z
+      .string()
+      .min(8, { message: t("password_length_validation") })
+      .regex(/\d/, { message: t("password_number_validation") })
+      .regex(/[a-z]/, {
+        message: t("password_lowercase_validation"),
+      })
+      .regex(/[A-Z]/, {
+        message: t("password_uppercase_validation"),
+      }),
+    new_password_2: z
+      .string()
+      .min(1, { message: t("please_enter_confirm_password") }),
+  })
+  .refine((values) => values.new_password_1 === values.new_password_2, {
+    message: t("password_mismatch"),
+    path: ["new_password_2"],
+  })
+  .refine((values) => values.new_password_1 !== values.old_password, {
+    message: t("new_password_same_as_old"),
+    path: ["new_password_1"],
+  });
 
 export default function UserResetPassword({
   userData,
 }: {
   userData: UserBase;
 }) {
   const { t } = useTranslation();
-  const [isSubmitting, setisSubmitting] = useState(false);
   const [isEditing, setIsEditing] = useState(false);
-
-  const initForm: PasswordForm = {
-    username: userData.username,
-    old_password: "",
-    new_password_1: "",
-    new_password_2: "",
-  };
-
-  const validateNewPassword = (password: string) => {
-    if (
-      password.length < 8 ||
-      !/\d/.test(password) ||
-      password === password.toUpperCase() ||
-      password === password.toLowerCase()
-    ) {
-      return false;
-    }
-    return true;
-  };
-
-  const validateForm = (formData: PasswordForm) => {
-    const errors: Partial<Record<keyof PasswordForm, string>> = {};
-
-    if (!formData.old_password) {
-      errors.old_password = t("please_enter_current_password");
-    }
-
-    if (!formData.new_password_1) {
-      errors.new_password_1 = t("please_enter_new_password");
-    } else if (!validateNewPassword(formData.new_password_1)) {
-      errors.new_password_1 = t("new_password_validation");
-    }
-
-    if (!formData.new_password_2) {
-      errors.new_password_2 = t("please_confirm_password");
-    } else if (formData.new_password_1 !== formData.new_password_2) {
-      errors.new_password_2 = t("password_mismatch");
-    }
-
-    if (formData.new_password_1 === formData.old_password) {
-      errors.new_password_1 = t("new_password_same_as_old");
-    }
-
-    return errors;
-  };
-
-  const handleSubmit = async (formData: PasswordForm) => {
-    setisSubmitting(true);
+  const [isPasswordFieldFocused, setIsPasswordFieldFocused] = useState(false);
+
+  const resetPasswordForm = useForm({
+    resolver: zodResolver(PasswordSchema),
+    defaultValues: {
+      old_password: "",
+      new_password_1: "",
+      new_password_2: "",
+    },
+  });
+  const { mutate: resetUserPasswordMutate, isPending } = useMutation({
+    mutationFn: mutate(routes.updatePassword),
+    onSuccess: (data: any) => {
+      toast.success(data?.message as string);
+      resetPasswordForm.reset();
+    },
```

Comment:
don't use `any` type. use types infered by the routes schema
```suggestion
    onSuccess: (data) => {
      toast.success(data?.message as string);
      resetPasswordForm.reset();
    },
```